src/oph/heratepalvelu/common.clj

```diff
@@ -370,6 +367,30 @@
   (and (> (:status (ex-data e)) 399)
        (< (:status (ex-data e)) 500)))
 
+(defn get-tila
+  "Hakee opiskeluoikeuden tilan tiettynä päivänä."
+  ([opiskeluoikeus vahvistus-pvm mode]
+   (let [offset (if (= mode :one-day-offset) 1 0)
+         jaksot (sort-by :alku (:opiskeluoikeusjaksot (:tila opiskeluoikeus)))]
+     (-> (partition 2 1 jaksot)
+         (->> (some (fn [[current next]]
+                      (and (< (compare vahvistus-pvm (:alku next)) offset)
+                           current))))
```

Comment:
Omasta mielestäni tuo aikaisempi `reduce` ei ollut mitenkään ei-idiomaattinen tässä kohtaa, ehkä jopa hivenen selkeämpi, ja tuo `reduced` hoitaa short-circuittaamisen. Välttyy myös noiden erityyppisten threading-makrojen sisäkkäin laittamiselta.